docs/Design/numerical-types.md

```diff
@@ -0,0 +1,134 @@
+# F´ Numerical Types
+
+On physical hardware, numerical types have a size in bits. The sizes of these integers are either explicitly set by the
+programmer or implicitly set by the compiler. i.e. `uint32_t` is explicitly 32-bits whereas `int` is implicitly set by
+the compiler. Flight software practice encourages use of explicitly sized types because there are fewer mistake made when the
+programmer uses an integer of a known size. However, in-practice it is nice to also use semantic types that represent
+concepts in their system (e.g. size) and are set by the platform supporter to a known size.
+
+This document describes: fixed-width types and semantic types.
+
+## Fixed Width Types
+
+In F´, fixed width types map to the standard definitions either in the C standard or in the `stdint.h` header as seen
```

Comment:
Clarify here what "turn on and off" actually means here. Explain how the numeric typedefs get turned off. Maybe link to the BasicTypes.hpp file where the types are enabled or disabled. "Turn off types that your compiler does not support"